test/framework/src/main/java/org/elasticsearch/test/transport/AssertingLocalTransport.java

```diff
@@ -1,110 +0,0 @@
-import org.elasticsearch.transport.TransportResponse;
-import org.elasticsearch.transport.TransportResponseHandler;
-import org.elasticsearch.transport.local.LocalTransport;
-
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.Random;
-import java.util.function.Supplier;
-
-public class AssertingLocalTransport extends LocalTransport {
-
-    public static final String ASSERTING_TRANSPORT_NAME = "asserting_local";
-
-    public static class TestPlugin extends Plugin implements NetworkPlugin {
-
-        @Override
-        public List<Setting<?>> getSettings() {
-            return Arrays.asList(ASSERTING_TRANSPORT_MIN_VERSION_KEY, ASSERTING_TRANSPORT_MAX_VERSION_KEY);
-        }
-
-        @Override
-        public Map<String, Supplier<Transport>> getTransports(Settings settings, ThreadPool threadPool, BigArrays bigArrays,
-                                                              CircuitBreakerService circuitBreakerService,
-                                                              NamedWriteableRegistry namedWriteableRegistry,
-                                                              NetworkService networkService) {
-            return Collections.singletonMap(ASSERTING_TRANSPORT_NAME,
-                () -> new AssertingLocalTransport(settings, circuitBreakerService, threadPool, namedWriteableRegistry));
-        }
-    }
-
-    public static final Setting<Version> ASSERTING_TRANSPORT_MIN_VERSION_KEY =
-        new Setting<>("transport.asserting.version.min", Integer.toString(Version.CURRENT.minimumCompatibilityVersion().id),
-            (s) -> Version.fromId(Integer.parseInt(s)), Property.NodeScope);
-    public static final Setting<Version> ASSERTING_TRANSPORT_MAX_VERSION_KEY =
-        new Setting<>("transport.asserting.version.max", Integer.toString(Version.CURRENT.id),
-            (s) -> Version.fromId(Integer.parseInt(s)), Property.NodeScope);
-    private final Random random;
-    private final Version minVersion;
-    private final Version maxVersion;
-
-    @Inject
-    public AssertingLocalTransport(Settings settings, CircuitBreakerService circuitBreakerService, ThreadPool threadPool,
-                                   NamedWriteableRegistry namedWriteableRegistry) {
-        super(settings, threadPool, namedWriteableRegistry, circuitBreakerService);
-        final long seed = ESIntegTestCase.INDEX_TEST_SEED_SETTING.get(settings);
-        random = new Random(seed);
-        minVersion = ASSERTING_TRANSPORT_MIN_VERSION_KEY.get(settings);
-        maxVersion = ASSERTING_TRANSPORT_MAX_VERSION_KEY.get(settings);
-    }
-
-    @Override
-    protected void handleParsedResponse(final TransportResponse response, final TransportResponseHandler handler) {
-        ElasticsearchAssertions.assertVersionSerializable(VersionUtils.randomVersionBetween(random, minVersion, maxVersion), response,
```

Comment:
Are we ok with dropping this assertion and the one in the method below? Should we add them to `MockTransportService`?